virtual_ecosystem/models/animal/animal_cohorts.py

```diff
@@ -931,213 +932,106 @@ def delta_mass_predation(
 
         return total_consumed_mass
 
-    def calculate_consumed_mass_herbivory(
+    def default_consumed_resource_mass(
         self,
-        plant_list: list[Resource],
-        target_plant: Resource,
+        resource_list: list[Resource],
+        target: Resource,
         adjusted_dt: timedelta64,
     ) -> float:
-        """Calculates the mass to be consumed from a plant resource by the herbivore.
-
-        This method utilizes the F_i_k method to determine the rate at which the target
-        plant is consumed, and then calculates the actual mass to be consumed based on
-        this rate and other model parameters.
+        """Standard search/handling time consumption using F_i_k (non-predation).
 
         Args:
-            plant_list: A list of plant resources that can be consumed by the
-                herbivore.
-            target_plant: The plant resource from which mass will be consumed.
-            adjusted_dt: The amount of time (D) in the time-step available for foraging.
+            resource_list: List of resource objects (e.g. litter, plants, etc.).
+            target: A specific resource from which biomass is being consumed.
+            adjusted_dt: Time available for foraging.
 
         Returns:
-            The mass to be consumed from the target plant by the herbivore (in kg).
+            Mass (kg) to consume from target.
         """
-        F = self.F_i_k(plant_list, target_plant)  # Adjusting this call as necessary
-
-        consumed_mass = target_plant.mass_current * (1 - exp(-(F * adjusted_dt)))
-        return consumed_mass
+        F = self.F_i_k(resource_list, target)
+        consumed = target.mass_current * (1.0 - exp(-F * adjusted_dt))
+        return max(consumed, 0.0)
```

Comment:
If `F` is positive (I assume `adjusted_dt` is positive), then `consumed` can not be negative, so the `max(consumed, 0.0)` bit is not really necessary:

```suggestion
        return target.mass_current * (1.0 - exp(-F * adjusted_dt))
```